src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -215,8 +215,22 @@ private async Task InitializeMEFAsync()
                 await DeleteOnRestartManager.Value.DeleteMarkedPackageDirectoriesAsync(ProjectContext.Value);
             }
 
-            ProjectRetargetingHandler = new ProjectRetargetingHandler(_dte, SolutionManager.Value, this, componentModel);
-            ProjectUpgradeHandler = new ProjectUpgradeHandler(this, SolutionManager.Value);
+            var vsTrackProjectRetargeting = await GetServiceAsync(typeof(SVsTrackProjectRetargeting)) as IVsTrackProjectRetargeting;
```

Comment:
I really don't understand why people use this `something as IWhatever` to do casting. If it fails, not matter how unlikely, you'll get a null value and then the program crashes with a null reference exception. If you use "normal" casting syntax, you'll get a clear exception saying that "type1" can't be cast to "type2".

Additionally, there's a extension method `GetServiceAsync<T1, T2>` that returns type `T2`, so you don't need to cast.

Reply:
Ok. Addressed it.